purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1235,7 +1241,7 @@ class Purchases internal constructor(
                             )
                         }
                     )
-                } else {
+                } ?: run {
```

Comment:
Hmm I'm wondering, will this case ever happen? As in, having a `StoreTransaction` without any `productId`... Seems like a bug to me and I would probably log an error, but not sure if I'm missing some valid case here.

Reply:
@tonidero Ha! I had an error here originally in this PR but then it looked like it changed the behavior as @swehner mentioned so I removed it 🤷‍♂️ 

I think the concern was with restores which is why I put this back to the way it is now 🤔

Reply:
Oh, I think this code does not get called for restores, but it does get called for pending purchases. But in both cases, I think we should still have a `productId` so I think this code shouldn't be reached... Again, not sure if I'm missing some case where this could be null but in `main`, it doesn't seem like that's a case we are handling right now https://github.com/RevenueCat/purchases-android/blob/main/purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt#L1153-L1178

Reply:
So should we changes this to verify `purchase.productIds` is not empty before trying to query store products? And if empty then throw an error?

Reply:
Well, it looks like we were not verifying that here at all in 5.6.x. Instead, when querying the productIds with Google, we would show a debug log: https://github.com/RevenueCat/purchases-android/blob/main/feature/google/src/unityIAP/java/com/revenuecat/purchases/google/BillingWrapper.kt#L149. 

So as long as we revert [this other change](https://github.com/RevenueCat/purchases-android/pull/721#discussion_r1055659367), IMO, we should just go back to that and not verify that case here at all.

Reply:
Reverted this back to sending up the entire list/set of product ids!